install_script.sh.template

```diff
@@ -795,50 +826,58 @@ fi
 
 # Use /usr/sbin/service by default.
 # Some distros usually include compatibility scripts with Upstart or Systemd. Check with: `command -v service | xargs grep -E "(upstart|systemd)"`
-restart_cmd="$sudo_cmd $service_cmd $system_service restart"
-stop_instructions="$sudo_cmd $service_cmd $system_service stop"
-start_instructions="$sudo_cmd $service_cmd $system_service start"
-
-if [[ `$sudo_cmd ps --no-headers -o comm 1 2>&1` == "systemd" ]] && command -v systemctl 2>&1; then
-  # Use systemd if systemctl binary exists and systemd is the init process
-  restart_cmd="$sudo_cmd systemctl restart ${system_service}.service"
-  stop_instructions="$sudo_cmd systemctl stop $system_service"
-  start_instructions="$sudo_cmd systemctl start $system_service"
-elif /sbin/init --version 2>&1 | grep -q upstart; then
-  # Try to detect Upstart, this works most of the times but still a best effort
-  restart_cmd="$sudo_cmd stop $system_service || true ; sleep 2s ; $sudo_cmd start $system_service"
-  stop_instructions="$sudo_cmd stop $system_service"
-  start_instructions="$sudo_cmd start $system_service"
-fi
-
-if [ $no_start ]; then
-    printf "\033[34m
-* DD_INSTALL_ONLY environment variable set: the newly installed version of the
-$nice_flavor will not be started. You will have to do it manually using the
-following command:
-
-    $start_instructions
-
-\033[0m\n"
-    exit
-fi
-
-printf "\033[34m* Starting the $nice_flavor...\n\033[0m\n"
-eval "$restart_cmd"
-
-
-# Metrics are submitted, echo some instructions and exit
-printf "\033[32m
-
-Your $nice_flavor is running and functioning properly. It will continue
-to run in the background and submit metrics to Datadog.
-
-If you ever want to stop the $nice_flavor, run:
-
-    $stop_instructions
-
-And to run it again run:
-
-    $start_instructions
+#
+declare -a services
+services=("$system_service")
+if [ -n "$fips_mode" ]; then
+  services+=("$system_fips_service")
+fi
+
+preamble="  * DD_INSTALL_ONLY environment variable set."
```

Comment:
I find this section hard to read (it uses the `preamble` text to only show some lines for the first service, assuming that the second service is the FIPS proxy).

I would be more explicit, and make the conditionals compare the `current_service` variable with the `datadog-fips-proxy` string:
- do not show the preamble text if `$current_service` is `datadog-fips-proxy`
- do not show the "It will continue to run in the background and submit metrics to Datadog." line if `$current_service` is `datadog-fips-proxy`.

WDYT?

Reply:
So the reason I set the preample up, is because I want it on the first iteration no matter what order we traverse the services in. If we add in the future a `datadog-vector` package that runs before, or we mess the services array order for whatever reason, the current logic would still behave correctly.

I'm willing to change it if you wish.

Reply:
I understand. My doubt is on whether what we're targeting is really the "first service", because right now the `preamble` variable changes the behavior of two different things. If in the future we install `datadog-vector` + `datadog-agent` + `datadog-fips-proxy` (in this order), then it's less clear to me which services should have the `* DD_INSTALL_ONLY environment variable set.` line, and which services should have the `It will continue to run in the background and submit metrics to Datadog.` line, and the answer may not be the "first service" for both.

To me:
- the `* DD_INSTALL_ONLY environment variable set.` should stay on the first iteration, so maybe it should just be printed before the loop for simplicity (one additional `if [ $no_start ]; then` conditional seems more readable to me than carrying that `preamble` variable logic).
- the `It will continue to run in the background and submit metrics to Datadog.` line would still apply to the `datadog-agent` service (or all services that actively fetch & send metrics, in the future), no matter what its position is, so I would explicit target (or exclude) the services that should (not) print that message.